examples/server/server.cpp

```diff
@@ -2967,11 +2967,18 @@ int main(int argc, char ** argv) {
     };
 
     const auto handle_props = [&ctx_server](const httplib::Request & req, httplib::Response & res) {
+        std::string template_key = "tokenizer.chat_template", curr_tmpl;
```

Comment:
I think it would be better to use `std::vector<char>` for `curr_tmpl`:

```cpp
std::vector<char> tmpl;
...
tmpl.resize(tlen);
llama_model_meta_val_str(ctx_server.model, template_key.c_str(), tmpl.data(), tmpl.size());
...
{ "chat_template", std::string(tmpl.data(), tmpl.size()) }
```